dbt/adapters/bigquery/impl.py

```diff
@@ -686,6 +686,29 @@ def load_dataframe(self, database, schema, table_name, agate_table,
         with self.connections.exception_handler("LOAD TABLE"):
             self.poll_until_job_completes(job, timeout)
 
+    @available.parse_none
+    def upload_file(self, local_file_path: str, database: str, table_schema: str,
+                    table_name: str, **kwargs) -> None:
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
+
+        table_ref = self.connections.table_ref(database, table_schema, table_name, conn)
+
+        load_config = google.cloud.bigquery.LoadJobConfig()
+        for k, v in kwargs['kwargs'].items():
+            if k == "schema":
+                setattr(load_config, k, json.loads(v))
+            else:
+                setattr(load_config, k, v)
+
+        with open(local_file_path, "rb") as f:
+            job = client.load_table_from_file(f, table_ref, rewind=True,
+                                              job_config=load_config)
+
+        timeout = self.connections.get_timeout(conn)
```

Comment:
@pgoslatara this `get_timeout` method was throwing an error for me when testing ex:`'BigQueryConnectionManager' object has no attribute 'get_timeout'` and was curious if you had a package installed that gave you access to it? I was able to replace it with the `get_job_execution_timeout_seconds` method as seen in `load_dataframe`

Reply:
Looks like my branch was outdated, it was missing this [commit](https://github.com/dbt-labs/dbt-bigquery/commit/141b86749df813cf3a3a90a90e7a7dfc401ba9b0). I've updated this method as you suggested, [commit](https://github.com/dbt-labs/dbt-bigquery/pull/121/commits/7b23a10890567cbad2cc4c4367c54f122895c24b).